services/galley/src/Galley/Data.hs

```diff
@@ -347,16 +347,22 @@ createConnectConversation a b name conn = do
 createOne2OneConversation :: U.UUID U.V4
                           -> U.UUID U.V4
                           -> Maybe (Range 1 256 Text)
+                          -> Maybe TeamId
                           -> Galley Conversation
-createOne2OneConversation a b name = do
+createOne2OneConversation a b name ti = do
     let conv = one2OneConvId a b
         a'   = Id (U.unpack a)
         b'   = Id (U.unpack b)
     now <- liftIO getCurrentTime
-    retry x5 $
-        write Cql.insertConv (params Quorum (conv, One2OneConv, a', privateOnly, fromRange <$> name, Nothing))
+    retry x5 $ case ti of
+        Nothing  -> write Cql.insertConv (params Quorum (conv, One2OneConv, a', privateOnly, fromRange <$> name, Nothing))
+        Just tid -> batch $ do
+            setType BatchLogged
+            setConsistency Quorum
+            addPrepQuery Cql.insertConv (conv, One2OneConv, a', privateOnly, fromRange <$> name, Just tid)
```

Comment:
Shouldn't these additional conversation Cassandra entries get deleted again if a team is deleted (or both of the users are deleted)?

Reply:
Yes, they get delete both when:
* [a team is deleted](https://github.com/wireapp/wire-server/blob/issue-581/services/galley/src/Galley/Data.hs#L193) - all team conversations are deleted
* [one of the users is deleted](https://github.com/wireapp/wire-server/blob/issue-581/services/galley/src/Galley/API/Internal.hs#L58) - that user is removed from the conversation